pagopa/io-pay-portal
# [IOPAY-146] error from nodo pagoPa responses + mixpanel events

This PR add the capability to manage the following semantic errors, related to verify and activation phase with `pagopa-proxy`: 
- PAYMENT_DUPLICATED
- INVALID_AMOUNT
- PAYMENT_ONGOING
- PAYMENT_EXPIRED
- PAYMENT_UNAVAILABLE
- PAYMENT_UNKNOWN

and mixpanel events

#### List of Changes


- Update `response.ts` to manage error internal response 
- add custom modal call on error response from `pagopa-proxy`
- add `mixpanel.track` event in to verify, activation and get payment with `pagopa-proxy`


#### Motivation and Context

To show a specific errore message to user as "Avviso già pagato"
See here to details [IOPAY-146](https://pagopa.atlassian.net/browse/IOPAY-146)

Add [mixpanel](https://mixpanel.com/) event , see [IOPAY-170](https://pagopa.atlassian.net/browse/IOPAY-170)



#### How Has This Been Tested?




_Manual test_
1. Run local `io-functions-pay-portal` typing `yarn start`
2. Run local `io-mock` typing 
> with custom and modify `io-pagopa-node-mock` ( using `docker-compose.override.yml` to do it )
3. Send a **GET** http://localhost:7071/api/v1/payment-requests/77234567891010001234567890123

To verify mixpanel event add `IO_PAY_PORTAL_ENV=develop` before starting frontend as follow :
`IO_PAY_PORTAL_ENV=develop yarn dev:proxy` 
in this way al `mixpanel.track` call are mock on `console.log` and could be inspect by _devtools_

#### Screenshots (if appropriate):

#### Types of changes




#### Checklist:
